services/catalog/rs_server_catalog/user_catalog.py

```diff
@@ -448,6 +452,16 @@ async def manage_put_post_request(self, request: Request) -> Request | JSONRespo
                 return JSONResponse(content=detail, status_code=HTTP_401_UNAUTHORIZED)
 
             if request.scope["path"] == "/collections":
+                # Manage a collection creation. The apikey user (or local user if in local mode)
+                # should be the same as the owner field in the body request. In other words, the
+                # apikey user cannot create a collection owned by another user
+                if user != user_login:
+                    detail = {
+                        "error": f"The '{user_login}' user cannot create a \
+collection owned by the '{user}' user",
+                    }
+                    logger.error(detail["error"])
+                    return JSONResponse(content=detail, status_code=HTTP_400_BAD_REQUEST)
```

Comment:
Status code should be one of 401, 403, **405**.

Reply:
Corrected